lib/sensors/optical_flow.cpp

```diff
@@ -0,0 +1,58 @@
+#include "optical_flow.hpp"
+
+namespace hyped::sensors {
+
+std::optional<OpticalFlow> OpticalFlow::create(core::ILogger &logger,
+                                               std::shared_ptr<io::ISpi> spi,
+                                               const std::uint8_t channel)
+{
+  // check we are communicating with the correct sensor
+  const std::uint8_t *device_id;
+  spi->read(kDeviceIdAddress, device_id, 1);
+  if (!device_id) {
+    logger.log(core::LogLevel::kFatal, "Failed to read the optical flow device");
+    return std::nullopt;
+  }
+  if (*device_id != kExpectedDeviceIdValue) {
+    logger.log(core::LogLevel::kFatal, "Failure, mismatched device ID for optical flow sensor");
+    return std::nullopt;
+  }
+  return OpticalFlow(logger, spi, channel);
+}
+
+OpticalFlow::OpticalFlow(core::ILogger &logger,
+                         std::shared_ptr<io::ISpi> spi,
+                         const std::uint8_t channel)
+    : logger_(logger),
+      spi_(spi),
+      channel_(channel)
+{
+}
+
+OpticalFlow::~OpticalFlow()
+{
+}
+
+// inspired by Bitcraze_PMW3901 on github
+// (https://github.com/bitcraze/Bitcraze_PMW3901/blob/master/src/Bitcraze_PMW3901.cpp)
+std::uint8_t OpticalFlow::getDeltaX(std::shared_ptr<io::ISpi> spi)
+{
+  std::uint8_t *x_low;
+  std::uint8_t *x_high;
+
+  spi->read(kXLowAddress, x_low, 1);
+  spi->read(kXHighAddress, x_high, 1);
+
+  return static_cast<std::int16_t>((*x_high << 8) | *x_low);
```

Comment:
check that this actually works, it might shift `x_high` by 8 which 0s it out before the cast applies